lua/lualine/components/grapple.lua

```diff
@@ -32,9 +32,21 @@ function Component:update_status()
 
     local current = grapple.find({ buffer = 0 })
 
+    local app_ok, app = pcall(require, "grapple.app")
+    local quick_select = nil
+    if app_ok then
+        quick_select = app.get().settings:quick_select()
+    end
+
     local output = {}
     for i, tag in ipairs(tags) do
-        local tag_str = tag.name and tag.name or i
+        local tag_str = tostring(i)
+        if tag.name then
+            tag_str = tag.name
+        elseif quick_select and quick_select[i] then
+            tag_str = quick_select[i]
+        end
```

Comment:
No need to cast to a string, and the above suggestion means don't need to do a first nil check on quick_select.

```suggestion
        -- stylus: ignore
        local tag_str = tag.name and tag.name
            or quick_select[i] and quick_select[i]
            or i
```